### ISSUE TITLE
Inconsistency in SDK check in NetworkObserver.kt

### ISSUE THREAD

**Issue by maharshpatel-wrk** [PierfrancescoSoffritti/android-youtube-player#995]
Here check is for versions greater than LOLIPOP while calling destroy. 

https://github.com/PierfrancescoSoffritti/android-youtube-player/blob/3b17fe9b69398bae570392b34caa1c234dcbaa28/core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/utils/NetworkObserver.kt#L39

but while initializing there is a check for N.

https://github.com/PierfrancescoSoffritti/android-youtube-player/blob/3b17fe9b69398bae570392b34caa1c234dcbaa28/core/src/main/java/com/pierfrancescosoffritti/androidyoutubeplayer/core/player/utils/NetworkObserver.kt#L30

Is this intentional?

**Comment by PierfrancescoSoffritti** [comment]
I can't remember but probably. Easily verifiable by checking the min api of those methods.

**Comment by maharshpatel247** [comment]
Yes, I checked those methods but there can be unexpected behaviors in Lolipop and Marshmellow SDKs as they will not unregister the broadcast receivers.

**Comment by PierfrancescoSoffritti** [comment]
I see, feel free to send a PR so we can discuss there :)
